src/VisualStudio/Core/Def/Indentation/VisualStudioInferredIndentationService.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Indentation;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.VisualStudio.LanguageServices.Indentation
+{
+    [ExportWorkspaceService(typeof(IInferredIndentationService), ServiceLayer.Host), Shared]
```

Comment:
I've never done this type of override before, please let me know if I did it wrong :).

Reply:
This is the correct way to export a service override for Visual Studio.